plugins/inputs/dcos/dcos.go

```diff
@@ -0,0 +1,453 @@
+  # List of  DC/OS agent hostnames from which the metrics should be gathered. Leave empty for all.
+  agents = []
+  # DC/OS agent node file system mount for which related metrics should be gathered. Leave empty for all.
+  file_system_mounts = []
+  # DC/OS agent node network interface names for which related metrics should be gathered. Leave empty for all.
+  network_interfaces = []
+  # HTTP Response timeout, value must be more than a second
+  #client_timeout = 4s
+`
+
+func (m *Dcos) Description() string {
+	return "Input plugin for gathering DC/OS agent metrics"
+}
+
+func (m *Dcos) SampleConfig() string {
+	return sampleConfig
+}
+
+var snakeCaser = strings.NewReplacer(".", "_")
+
+type mesosMasterStateSummary struct {
+	Hostname string  `json:"hostname"`
+	Slaves   []slave `json:"slaves"`
+}
+
+type mesosAgentContainer struct {
+	Id string `json:"container_id"`
+}
+
+type slave struct {
+	Id       string `json:"id"`
+	Hostname string `json:"hostname"`
+}
+
+type datapoint struct {
+	Name      string                 `json:"name"`
+	Value     interface{}            `json:"value"`
+	Unit      string                 `json:"unit"`
+	TimeStamp string                 `json:"timestamp"`
+	Tags      map[string]interface{} `json:"tags"`
+}
+
+type metric struct {
+	Datapoints []datapoint            `json:"datapoints"`
+	Dimensions map[string]interface{} `json:"dimensions"`
+}
+
+const (
+	Node      = "node"
+	Container = "container"
+	App       = "app"
+)
+
+const MesosMasterStateSummaryUrl = "/mesos/master/state-summary"
+const MesosSlaveContainersUrl = "/agent/%s/containers"
+const AgentContainersUrl = "/system/v1/agent/%s/metrics/v0/containers"
+const AgentContainerMetricsUrl = "/system/v1/agent/%s/metrics/v0/containers/%s"
+const AgentContainerAppMetricsUrl = "/system/v1/agent/%s/metrics/v0/containers/%s/app"
+const AgentNodeMetricsUrl = "/system/v1/agent/%s/metrics/v0/node"
+
+var tr *http.Transport
+var client *http.Client
+
+func getClient(timeout time.Duration) *http.Client {
+
+	if client == nil {
+		tr = &http.Transport{
+			ResponseHeaderTimeout: timeout - time.Second,
+		}
+		client = &http.Client{
+			Transport: tr,
+			Timeout:   timeout,
+		}
+	}
+	return client
+}
+
+//validateConfiguration tests whether important configuration params are not empty
+func (m *Dcos) validateConfiguration() error {
+	errorStrings := []string{}
+
+	if len(m.ClusterURL) == 0 {
+		errorStrings = append(errorStrings, "Invalid configuration, cluster_url is empty")
+	}
+	if len(m.AuthToken) == 0 {
+		errorStrings = append(errorStrings, "Invalid configuration, auth_token is empty")
+	}
+
+	if m.ClientTimeout.Duration.Seconds() == 0 {
+		m.ClientTimeout.Duration = time.Second * 4
+	} else if m.ClientTimeout.Duration.Seconds() <= 1 {
+		errorStrings = append(errorStrings, "Invalid configuration, timeout value must be greater than a second")
+	}
+
+	if len(errorStrings) > 0 {
+		return errors.New(strings.Join(errorStrings, "\n"))
+	} else {
+		return nil
+	}
```

Comment:
I would rather see all errors at once and correct them in a single shot, instead of returning to config several times, cause I missed sth in config and the app is complaining is still complaining

Reply:
Since we are sending this to the logfile we want to prefer single line messages.  Usability is not harmed significantly.